pkg/apis/openebs.io/lvm/v1alpha1/lvmsnapshot.go

```diff
@@ -29,8 +29,34 @@ type LVMSnapshot struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   VolumeInfo `json:"spec"`
-	Status SnapStatus `json:"status"`
+	Spec   LVMSnapshotSpec `json:"spec"`
+	Status SnapStatus      `json:"status"`
+}
+
+// LVMSnapshotSpec defines LVMSnapshot spec
+type LVMSnapshotSpec struct {
+	// OwnerNodeID is the Node ID where the snapshot group is present which is where
+	// the snapshot has been provisioned.
+	// OwnerNodeID can not be edited after the snapshot has been provisioned.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	OwnerNodeID string `json:"ownerNodeID"`
+
+	// ThinProvision specifies whether logical snapshots can be thinly provisioned.
+	// If it is set to "yes", then the LVM LocalPV Driver will create
+	// thinProvision i.e. logical snapshots that are larger than the available extents.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=yes;no
+	ThinProvision string `json:"thinProvision,omitempty"`
```

Comment:
this is not needed. Let us remove this one.

Reply:
I think this is good to have in lvmsnapshot spec, which can be also used to make the decision to create thin snapshot for snapshot create request.

Reply:
Prateek, we don't need any parameter. We can decide based on snapsize parameter. For thin provisioning, snapsize will be 
 empty. see this https://github.com/openebs/lvm-localpv/issues/98. For thick snapshot, snapsize is much parameter.